src/Cache/FileCacheStorage.php

```diff
@@ -28,20 +26,12 @@ public function __construct(private string $directory)
 
 	private function makeDir(string $directory): void
 	{
-		if (is_dir($directory)) {
+		if (DirectoryCreator::ensureDirectoryExists($directory, 0777)) {
 			return;
 		}
 
-		$result = @mkdir($directory, 0777);
-		if ($result === false) {
-			clearstatcache();
-			if (is_dir($directory)) {
-				return;
-			}
-
-			$error = error_get_last();
-			throw new InvalidArgumentException(sprintf('Failed to create directory "%s" (%s).', $this->directory, $error !== null ? $error['message'] : 'unknown cause'));
-		}
+		$error = error_get_last();
+		throw new InvalidArgumentException(sprintf('Failed to create directory "%s" (%s).', $this->directory, $error !== null ? $error['message'] : 'unknown cause'));
```

Comment:
Either we need to get rid of the specific error message here which relies on `error_get_last()` (leaks an implementation detail), or we need to throw a proper exception from DirectoryCreator, and surround each call with a try-catch.

I'd actually be in favour of the exception, because using return `bool` values to signal an error isn't great.

Reply:
Done